drivers/can/can_mcux_flexcan.c

```diff
@@ -1459,17 +1459,17 @@ static const struct can_driver_api mcux_flexcan_fd_driver_api = {
 		.clk_source = DT_INST_PROP(id, clk_source),		\
 		.bitrate = DT_INST_PROP(id, bus_speed),			\
 		.sjw = DT_INST_PROP(id, sjw),				\
-		.prop_seg = DT_INST_PROP_OR(id, prop_seg, 0),		\
-		.phase_seg1 = DT_INST_PROP_OR(id, phase_seg1, 0),	\
-		.phase_seg2 = DT_INST_PROP_OR(id, phase_seg2, 0),	\
+		.prop_seg = DT_INST_PROP_OR(id, prop_seg, 1),		\
+		.phase_seg1 = DT_INST_PROP_OR(id, phase_seg1, 1),	\
+		.phase_seg2 = DT_INST_PROP_OR(id, phase_seg2, 2),	\
 		.sample_point = DT_INST_PROP_OR(id, sample_point, 0),	\
 		IF_ENABLED(CONFIG_CAN_MCUX_FLEXCAN_FD, (		\
 			.flexcan_fd = DT_NODE_HAS_COMPAT(DT_DRV_INST(id), FLEXCAN_FD_DRV_COMPAT), \
 			.bitrate_data = DT_INST_PROP_OR(id, bus_speed_data, 0),		\
-			.sjw_data = DT_INST_PROP_OR(id, sjw_data, 0),			\
+			.sjw_data = DT_INST_PROP_OR(id, sjw_data, 1),			\
 			.prop_seg_data = DT_INST_PROP_OR(id, prop_seg_data, 0),		\
-			.phase_seg1_data = DT_INST_PROP_OR(id, phase_seg1_data, 0),	\
-			.phase_seg2_data = DT_INST_PROP_OR(id, phase_seg2_data, 0),	\
+			.phase_seg1_data = DT_INST_PROP_OR(id, phase_seg1_data, 1),	\
+			.phase_seg2_data = DT_INST_PROP_OR(id, phase_seg2_data, 2),	\
```

Comment:
The `_OR` part here is just for DTS nodes where this optional property is not defined. They should default to `0` as currently in the code.

Reply:
We do a `-1` in the below lines:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcux_flexcan.c#L1269-L1280

Default of `0` would put `-1` into `timingConfig`

Reply:
If the property is not set, the timing algorithm is used. So those values are not used.

Reply:
@alexanderwachter Thank you for your review. By looking at the code the timing algorithm is used for certain cases as seen by the below line:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcux_flexcan.c#L1189

However for the else cases it seems to use the config default values if these are not specified in device tree.

Reply:
Tle else case is used only if the sp is not set. This is where the rest of the values must be valid, hence the defaults are not used.